File: Assignment_03/Program5/Program5.c
////////////////////////////////////////////////////////////////////////////////
//
//  File name: Program5.c
//  Description: Used to check whether the character is a vowel or not.
//  Author: Parimal Kelkar
//  Date: 11/05/2025
//
////////////////////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<stdbool.h>

////////////////////////////////////////////////////////////////////////////////
//
//  Name: ChkVowel()
//  Description: Accepts a character and checks whether it is a vowel or not.
//  Input: char
//  Output: bool
//  Author: Parimal Kelkar
//  Date: 11/05/2025
//
////////////////////////////////////////////////////////////////////////////////
bool ChkVowel(char cValue)
{
    if(((cValue == 'A') || (cValue == 'a')) || ((cValue == 'E') || (cValue == 'e')) || ((cValue == 'I') || (cValue == 'i')) || ((cValue == 'O') || (cValue == 'o')) ||((cValue == 'U') || (cValue == 'u')))
    {
        return true;
    }
    else
    {
        return false;
    }
}

int main()
{
    char cValue = '\0';
    bool bRet = false;

    printf("Enter character:");
    scanf("%c", &cValue);

    bRet = ChkVowel(cValue);

    if(bRet == true)
    {
        printf("It is a vowel.\n");
    }
    else
    {
        printf("It is not a vowel.\n");
    }

    return 0;
}

File: Assignment_45/Program5/program5.java
////////////////////////////////////////////////////////////////////////////////
// 
// File: program5.java
// Description: Used to accept a matrix from user and check whether it is a sparse matrix or not.
// Author: Parimal Kelkar
// Date: 02/08/2025
// 
////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Matrix
{
    ////////////////////////////////////////////////////////////////////////////
    //
    // Method: ChkSparse()
    // Description: Accepts a matrix and checks whether it is a sparse matrix or not.
    // Input: int[][], int, int
    // Output: boolean
    // Author: Parimal Kelkar
    // Date: 02/08/2025
    //
    ////////////////////////////////////////////////////////////////////////////
    boolean ChkSparse(int Arr[][], int iRow, int iCol)
    {
        // Updators
        if(iRow < 0)
        {
            iRow = -iRow;
        }

        if(iCol < 0)
        {
            iCol = -iCol;
        }

        int i = 0, j = 0, iCount = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                if(Arr[i][j] == 0)
                {
                    iCount++;
                }
            }
        }

        return (iCount > ((iRow * iCol) / 2));
    }

    public void Display(int Arr[][])
    {
        int i = 0, j = 0;

        System.out.println("\nThe matrix is:");
        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                System.out.print(Arr[i][j] + "\t");
            }

            System.out.println("\n");
        }
    }
}

class program5
{
    public static void main(String args[])
    {
        Scanner sObj = new Scanner(System.in);

        int iValue1 = 0, iValue2 = 0;

        System.out.println("Enter number of rows: ");
        iValue1 = sObj.nextInt();

        System.out.println("Enter number of columns: ");
        iValue2 = sObj.nextInt();

        int Arr[][] = new int[iValue1][iValue2];

        System.out.println("Enter " + iValue1 + " X " + iValue2 + " elements:");

        int no = 0;
        for(int i = 0; i < Arr.length; i++)
        {
            for(int j = 0; j< Arr[i].length; j++)
            {
                System.out.print("Enter element [" + i + "][" + j + "]: ");
                Arr[i][j] = sObj.nextInt();
            }
        }

        Matrix mObj = new Matrix();

        mObj.Display(Arr);

        boolean bRet = false;

        bRet = mObj.ChkSparse(Arr, iValue1, iValue2);

        if(true == bRet)
        {
            System.out.println("It is a sparse matrix.");
        }
        else
        {
            System.out.println("It is not a sparse matrix.");            
        }
    }
}